component/metrics/remotewrite/remote_write.go

```diff
@@ -174,15 +174,15 @@ func (c *Component) Receive(ts int64, metricArr []*metrics.FlowMetric) {
 	app := c.storage.Appender(context.Background())
 	for _, m := range metricArr {
 		// TODO this should all be simplified into one call
-		if m.GlobalRefID == 0 {
-			globalID := metrics.GlobalRefMapping.GetOrAddGlobalRefID(m.Labels)
-			m.GlobalRefID = globalID
+		if m.GlobalRefID() == 0 {
+			globalID := metrics.GlobalRefMapping.GetOrAddGlobalRefID(m.LabelsCopy())
+			m.SetGlobalRefID(globalID)
 		}
-		localID := metrics.GlobalRefMapping.GetLocalRefID(c.opts.ID, m.GlobalRefID)
-		newLocal, err := app.Append(storage.SeriesRef(localID), m.Labels, ts, m.Value)
+		localID := metrics.GlobalRefMapping.GetLocalRefID(c.opts.ID, m.GlobalRefID())
+		newLocal, err := app.Append(storage.SeriesRef(localID), m.LabelsCopy(), ts, m.Value())
```

Comment:
This seems a bit like a code smell. If the intent is that senders should generate their own ref IDs, then we should rely on that here and never mutate the in-flight samples. 

Plus, if the metrics are in-flight and they were generated ad-hoc from the sender, it's not going to get the updated ref ID as it is. 

I think this should just be 

```go
localID := metrics.GlobalRefMapping.GetOrAddGlobalRefID(m.Labels()) 
```

Reply:
Simplified this code path